src/ansys/mapdl/core/mapdl_grpc.py

```diff
@@ -2149,105 +2180,177 @@ def download(
+                )
+            list_files = []
+            for each in files:
+                list_files.extend(self._validate_files(each, extension=extension))
+
         else:
-            target_dir = os.getcwd()
+            raise ValueError(
+                f"The `file` parameter type ({type(files)}) is not supported."
+                "Only strings, tuple of strings or list of strings are allowed."
+            )
 
         for each_file in list_files:
-            try:
-                file_name = os.path.basename(
-                    each_file
-                )  # Getting only the name of the file.
-                #  We try to avoid that when the full path is supplied, it will crash when trying
-                # to do `os.path.join(target_dir"os.getcwd()", file_name "full filename path"`
-                # This will produce the file structure to flat out, but it is find, because recursive
-                # does not work in remote.
-                self._download(
-                    each_file,
-                    out_file_name=os.path.join(target_dir, file_name),
-                    chunk_size=chunk_size,
-                    progress_bar=progress_bar,
-                )
-            except FileNotFoundError:
-                # So far the grpc interface returns size of the file equal
-                # zero, if the file does not exists or its size is zero,
-                # but they are two different things!
-                # In theory, since we are obtaining the files name from
-                # `mapdl.list_files()` they do exist, so
-                # if there is any error, it means their size is zero.
-                pass  # this is not the best.
+            self._download(
+                each_file,
+                out_file_name=os.path.join(target_dir, each_file),
+                chunk_size=chunk_size,
+                progress_bar=progress_bar,
+            )
+
+        return list_files
+
+    def _validate_files(
+        self, file: str, extension: Optional[str] = None, recursive: bool = True
+    ) -> List[str]:
+        if extension is not None:
+            if not isinstance(extension, str):
+                raise TypeError(f"The extension {extension} must be a string.")
+
+            if extension[0] != ".":
+                extension = "." + extension
+
+        else:
+            extension = ""
+
+        if self.is_local:
+            # filtering with glob (accepting *)
+            if not os.path.dirname(file):
+                file = os.path.join(self.directory, file)
+            list_files = glob.glob(file + extension, recursive=recursive)
+
+        else:
+            base_name = os.path.basename(file + extension)
+            self_files = self.list_files()
+
+            list_files = fnmatch.filter(self_files, base_name)
+
+        # filtering by extension
+        list_files = [each for each in list_files if each.endswith(extension)]
```

Comment:
if `each` represent a file, you may want to name it `file`
```suggestion
        list_files = [file for file in list_files if file.endswith(extension)]
```